openvariant/variant/variant.py

```diff
@@ -37,6 +37,14 @@ def _open_file(file_path: str, mode='r+b'):
 
     return mm, file
 
+def _detect_delimiter(line: str):
+    """Detects the dominant delimiter in a line"""
+    counts = {
+        '\t': line.count('\t'),
+        ',': line.count(','),
+        ';': line.count(';')
+    }
+    return max(counts, key=counts.get)
```

Comment:
I was thinking of a more robust approach which involves the (already imported) csv library. We can use the csv sniffer class in order to infer in a more consistent manner the delimiter. This method automatically takes into account information about how to handle double quotes, whitespace and delimiters in general. Sources: [csv.Sniffer](https://docs.python.org/3/library/csv.html#csv.Sniffer), [csv.Dialect](https://docs.python.org/3/library/csv.html#csv.Dialect)

It could look something like:

```suggestion
def _detect_delimiter(line: str):
    sniffer = csv.Sniffer()
    dialect = sniffer.sniff(line, delimiters='\t,;')
    return dialect.delimiter
```

Reply:
Works great!